### ISSUE TITLE
Generated Polkadot SDK Templates Emit Errors

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/polkadot-sdk#5226]
When compiling the `minimal-template`, the following errors appear:

```
➜  minimal-template git:(master) cargo build --release
warning: unexpected `cfg` condition value: `try-runtime`
  --> pallets/template/src/lib.rs:20:12
   |
20 |     #[pallet::pallet]
   |               ^^^^^^
   |
   = note: expected values for `feature` are: `default` and `std`
   = help: consider adding `try-runtime` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
   = note: `#[warn(unexpected_cfgs)]` on by default

warning: `pallet-minimal-template` (lib) generated 1 warning
warning: unexpected `cfg` condition value: `runtime-benchmarks`
  --> node/src/service.rs:30:7
   |
30 | #[cfg(feature = "runtime-benchmarks")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expected values for `feature` are: `default`
   = help: consider adding `runtime-benchmarks` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
   = note: `#[warn(unexpected_cfgs)]` on by default

warning: unexpected `cfg` condition value: `runtime-benchmarks`
  --> node/src/service.rs:34:11
   |
34 | #[cfg(not(feature = "runtime-benchmarks"))]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expected values for `feature` are: `default`
   = help: consider adding `runtime-benchmarks` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

warning: `minimal-template-node` (lib) generated 2 warnings
warning: unexpected `cfg` condition value: `try-runtime`
  --> node/src/command.rs:26:7
   |
26 | #[cfg(feature = "try-runtime")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expected values for `feature` are: `default`
   = help: consider adding `try-runtime` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
   = note: `#[warn(unexpected_cfgs)]` on by default

warning: unexpected `cfg` condition value: `runtime-benchmarks`
  --> node/src/service.rs:30:7
   |
30 | #[cfg(feature = "runtime-benchmarks")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expected values for `feature` are: `default`
   = help: consider adding `runtime-benchmarks` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

warning: `minimal-template-node` (bin "minimal-template-node") generated 3 warnings (1 duplicate)
```

This is because the templates lack the `runtime-benchmarks` or `try_runtime` features, but these features are being referenced within the `service.rs` file and `command.rs` file included in the template.

We need to either change the service file to exclude these feature gated parts of the code, or (probably better) include these features into the generated templates.

**Comment by shawntabrizi** [comment]
cc @rzadp @kianenigma @mordamax

**Comment by rzadp** [comment]
As a quick measure (to get rid of compilation errors) [I proposed to remove](https://github.com/paritytech/polkadot-sdk/pull/5237) the unused code.

And looks like the error from `#[pallet::pallet]` will be solved by https://github.com/paritytech/polkadot-sdk/pull/5099

@kianenigma Do you think we should eventually add some benchmarking or try-runtiming functionality to the minimal template, or not in order to keep it minimal?

**Comment by shawntabrizi** [comment]
@rzadp my take on your question is that we should aim to keep the minimal template minimal, and then using this as a starting point, see how we can simplify the process to add benchmarks and try-runtime to a minimal template.

Perhaps this would require more macro magic, or some tooling.

Maybe we cannot succeed, and we have to put them in the template, but I think that should not be the first choice.

**Comment by shawntabrizi** [comment]
closing this issue because it is resolved with #5237
